### DISCUSSION TITLE
Which models in asreview are probabilistic or deterministic?

### DISCUSSION THREAD

**Discussion by DBergman1997** [asreview/asreview#984] [Q&A] [answered]
I was wondering which models in asreview are probabilistic and which are deterministic. 
We know the model Naive Bayes is probabilistic, and we are specifically curious about the models Logistic Regression and Neural Network.

That being said, we're also curious what category the other models fall into.

Thanks in advance! 😄

**Comment by bagheria** [comment, marked as answer]
Thanks @DBergman1997 for the interesting question.

Using this link (https://asreview.readthedocs.io/en/latest/_modules/index.html), you can read the complete documentation of all modules in ASReview. In the branch asreview.models.classifiers, we already have logistic regression, LSTM recurrent neural network, naïve Bayes, random forest, perceptron neural network, and support vector machine.

Regarding your question about probabilistic classifiers in ASReview, we have SVM as deterministic and the rest are probabilistic, while for every probabilistic method you can simply return the class with the highest probability and therefore will seem deterministic. Also, the other way around, based on the distance to the separating hyperplane in SVMs a probability can be computed and returned for each class. To this end, of course, you can fork the ASReview repo, and send a pull request.
